### DISCUSSION TITLE
How do I remove #?

### DISCUSSION THREAD

**Discussion by textpain** [Day8/re-frame-template#152] [Q&A] [answered]
Hello again. 

It seems like this boilerplate uses hash router. The '#' is always present at the address bar. I have considered todomvc example that uses proper routing (without '#'), but when I try to use it with figwheel, it doesn't work and it says "Keep on figwheelin'". 

So how do I remove # for both dev mode and production, the right way?

**Comment by gadfly361** [comment, marked as answer]
@uyouthe Here is a nice blog post by @pupeno that goes over how to remove the hashes using bidi and pushy
 https://pupeno.com/2015/08/26/no-hashes-bidirectional-routing-in-re-frame-with-bidi-and-pushy/

I am going to close this issue since it isn't a bug. However, if you want more help, check out re-frame [Clojure Slack channel](http://clojurians.net) or the [ClojureScript mailing list](https://groups.google.com/forum/#!forum/clojurescript).

**Comment by textpain** [comment]
Oh, I'm sorry. I didn't knew re-frame's slack exists. Thank you.

**Comment by gadfly361** [comment]
Oh, no need to apologize! I do recommend joining the slack group though, the #re-frame channel is always buzzing and everyone is very friendly :)
